ledger-api/grpc-definitions/com/daml/ledger/api/v1/event.proto

```diff
@@ -57,13 +57,20 @@ message CreatedEvent {
   Value contract_key = 7;
 
   // The arguments that have been used to create the contract.
-  // Set if there was a match by template_id in the given transaction filter.
+  // Set either:
+  //   - if there was a party, which is among the witnesses_parties of this event,
+  //     and for which an InclusiveFilters exists with the template_id of this event
+  //     among the event_ids,
+  //   - or if there was a party, which is among the witnesses_parties of this event,
+  //     and for which a wildcard filter exists (Filters without InclusiveFilters).
```

Comment:
from what I remember (and a quick test seems to confirm that). Not setting the inclusive filter is equivalent to setting it with an empty list of template ids so this is slightly imprecise.

Reply:
I am ahappy to correct, good catch.
Do you think the same should apply, as we will have interfaceFilters too? (like both of them empty?)
Is this a protobuf trickery or we make the mapping? :thinking:

Reply:
It’s not protobuf trickery. We actually map them to different values during [filter validation](https://github.com/digital-asset/daml/blob/ab948dd8de030cd54b809cee907caa439ad34ed7/ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/validation/TransactionFilterValidator.scala#L43) and then somewhere later (haven’t tracked it through the code) they seem to result in the same response again.

I think I don’t understand your suggestion wrt to interfaceFilters. We have to keep the current behavior for inclusive not set & template ids empty/not set. What are you suggestiong to change/extend?

Reply:
For completeness: We are doing as resolving both cases to an empty Set in the FilterRelation.
Shall we continue with this pattern? (like if empty list for template_ids and empty list for interface_filters equals to non-specified InclusiveFilters?)
@cocreature @meiersi-da

Reply:
> I think I don’t understand your suggestion wrt to interfaceFilters. We have to keep the current behavior for inclusive not set & template ids empty/not set. What are you suggestiong to change/extend?

I don't think we can keep it, in it's original form. How about the case if there is a InclusiveFilters defined with empty template_ids, and non-empty interface_filters? I assumed so far that this will boil down to no wildcard filter, right?

Reply:
> Shall we continue with this pattern? (like if empty list for template_ids and empty list for interface_filters equals to non-specified InclusiveFilters?)

Yeah, I’d keep that.

> I don't think we can keep it, in it's original form. How about the case if there is a InclusiveFilters defined with empty template_ids, and non-empty interface_filters? I assumed so far that this will boil down to no wildcard filter, right?

Sorry I was imprecise, by "current behavior" I meant if template_ids is not set and interface_filters is not set (which was always the case up until now), we treat it as a wildcard filter. If you do set interface_filters then it’s not a wildcard filter anymore.

Reply:
This is reflected now in the changes